src/backend/decorators/resource/resource-decorator.ts

```diff
@@ -184,11 +184,7 @@ class ResourceDecorator {
       const property = this.properties[propertyName]
 
       const subProperties = flatSubProperties(property)
-      return {
-        ...memo,
-        [propertyName]: property.toJSON(),
-        ...subProperties,
-      }
+      return Object.assign(memo, { [propertyName]: property.toJSON() }, subProperties)
```

Comment:
I changed it to Object.assign because it's much faster than spread according to this test and it still looks clean: https://www.measurethat.net/Benchmarks/Show/2978/0/lodash-merge-vs-objectassign-vs-spread

`getFlattenProperties` may affect performance because it's used in `toJSON` method which is frequently called everywhere.

Reply:
yep - spread creates a new object, object assign doesn't (unless you put {} as the first argument)